eng/build-job.yml

```diff
@@ -171,6 +178,13 @@ jobs:
           env:
             # TODO: remove NUGET_PACKAGES once https://github.com/dotnet/arcade/issues/1578 is fixed
             NUGET_PACKAGES: $(Build.SourcesDirectory)\.packages 
+        # Save nuget packages in pipeline for update dotnet/versions
+        # TODO remove task when https://github.com/dotnet/arcade/issues/1818 is fixed
+        - task: PublishPipelineArtifact@0
+          displayName: Save packages as pipeline artifact
+          inputs:
+            artifactName: ${{ format('build_{0}_{1}_{2}_nuget', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
+            targetPath: $(Build.SourcesDirectory)\bin\Product\$(osGroup).$(archType).$(buildConfigUpper)\.nuget\pkg
```

Comment:
Is the step duplication necessary? In my experience `/` works fine on Windows for this kind of thing, otherwise they look the same.